File: 2017/d11/sol.rb
input = File.read('input.txt').split(',')

x = 0
y = 0
z = 0
max = 0

input.each do |move|
  case move
  when 'n'
    y = y + 1
    z = z - 1
  when 'ne'
    x = x + 1
    z = z - 1
  when 'nw'
    x = x - 1
    y = y + 1
  when 's'
    y = y - 1
    z = z + 1
  when 'se'
    x = x + 1
    y = y - 1
  when 'sw'
    x = x - 1
    z = z + 1
  else
    raise "wtf: #{move}"
  end

  max = [max, ([x.abs, y.abs, z.abs].max)].max
end

puts max
puts [x.abs, y.abs, z.abs].max

File: 2018/1/sol.exs
File.read!("input.txt")
  |> String.trim
  |> String.split("\n", trim: true)
  |> Enum.map(&String.to_integer/1)
  |> Stream.cycle()
  |> Enum.reduce_while({0, MapSet.new([0])}, fn(x, {val, seen}) -> 
      newVal = val + x

      if MapSet.member?(seen, newVal) do
        {:halt, newVal}
      else
        {:cont, {newVal, MapSet.put(seen, newVal)}}
      end
    end)
  |> IO.puts